packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -420,8 +427,18 @@ export class AgentSchedulerFactory implements IFluidDataStoreFactory {
 
     public static async createChildInstance(parentContext: IFluidDataStoreContext): Promise<AgentScheduler> {
         const packagePath = [...parentContext.packagePath, AgentSchedulerFactory.type];
-        const router = await parentContext.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<AgentScheduler>(router, "/");
+        const dataStore = await parentContext.containerRuntime.createDataStore(packagePath);
+        // TODO: IFluidLoadable is currently only exposed in the DataStore class, not the IDataStore interface,
+        // thus the discovery with FluidObject. Once entrypoints are exposed more directly this should be
+        // simplified.
+        const maybeIFluidLoadable: IDataStore & FluidObject<IProvideFluidLoadable> = dataStore;
```

Comment:
Why are we not exposing handle directly on IDataStore?
It would be awful to recommend our partners to do similar dance in their code base. If it's temporal state, then maybe we do not touch this class until we land on final state and our repo is an example of how to write code?

f I think about ideal usage, it feels like it should be something like that:

> const instance = IDataStore.entryPoint.get() as IAgentScheduler;

or maybe even this:

> const instance = IDataStore.getEntryPoint() as IAgentScheduler;

Reply:
> Why are we not exposing handle directly on IDataStore?

For now, to keep from exposing any of this "intermediate state" externally. Once we get to the final state of how this all should look (basically like one of your proposals above), it will make it to the public interfaces (`IDataStore`, `IFluidDataStoreRuntime`, maybe `IFluidDataStoreChannel`).

I made the changes in AgentScheduler under the premise of "let's start seeing our own code leverage the new functionality", even if it's clunky for a bit, so we can start validating that things work as expected. Maybe we keep changes like that in a separate (draft) PR so we at least can make sure CI is passing for them, and keep rebasing that branch as we move forward building the foundations?

Reply:
I like the approach where we raise first draft PR with final shape of APIs and usage examples, even if (compat) tests are not passing, as it gives clear message on direction. Then it's easier to have that discussion :)

I agree that we want to do a change here. But my feedback is about what form it takes in final state, i.e. next line (and not about the fact that it's optional or not even present on signature), which current is equivalent to

>         const entrypoint: (FluidObject<IAgentScheduler> & IProvideFluidLoadable) | undefined =
            await datastore.IFluidLoadable.handle.get();

This looks awful from usability standpoint. I believe this is much better:

> const instance = await datastore.getEntryPoint() as IAgentScheduler;

Getting from here to there is hard (another breaking change), so I read current change as we are not going to do it and are on the path to keep this really long construct. That's what I suggest changing :)

(nit: If we want IAgentScheduler to implement IProvideFluidLoadable, it's better to bake it into its signature, i.e. make implementer of IAgentScheduler promise this support, vs. user hope for its presence).